torch_xla/torch_xla.py

```diff
@@ -125,6 +129,8 @@ def foo2(x):
     elif hasattr(f, '__str__'):
       name = f.__str__()
 
+  current_id = uuid.uuid4().__str__()
```

Comment:
I thought about it a bit and find this is not very ideal. This way if we do
```
def f():
  xxx   

torch.compile(f)
torch.compile(f)
```
we will get 2 uuid. I think we should try to hash the passed in function pointer so we can dedup.

Reply:
I think we should use function pointer if it is not None (currently when it is used as a decorator with @, the fn will be none I think).